packages/caravan-fees/src/btcTransactionComponents.ts

```diff
@@ -222,8 +296,13 @@ export class BtcTxInputTemplate extends BtcTxComponent {
       txid: this._txid,
       vout: this._vout,
       value: this._amountSats.toString(),
+      sequence: this._sequence || RBF_SEQUENCE, // Default it to signal RBF if not provided
       prevTxHex: this._nonWitnessUtxo?.toString("hex"),
       witnessUtxo: this._witnessUtxo,
+      redeemScript: this._redeemScript,
+      witnessScript: this._witnessScript,
+      bip32Derivations:
+        this._bip32Derivations.length > 0 ? this._bip32Derivations : undefined,
```

Comment:
can't you do

```suggestion
        this._bip32Derivations || undefined,
```
would an empty array break things though?

Reply:
changes [done](https://github.com/caravan-bitcoin/caravan/pull/304/commits/28c3c416276d786fe8ebc7bc83f8a69fed9cdf37)